src/epcpm/sunspectocsv.py

```diff
@@ -0,0 +1,532 @@
+                )
+
+                row = Fields()
+
+                for name, field in attr.asdict(point_fields).items():
+                    if field is None:
+                        continue
+
+                    setattr(row, name, getattr(point, field.name))
+
+                row.address_offset = self.address_offset
+                row.name = table_element2.name
+                row.label = table_element2.abbreviation
+                row.type_uuid = row.type
+                row.type = self.parameter_uuid_finder(row.type).name
+                row.units = table_element2.units
+                row.modbus_address = (
+                    block_size + self.model_offset + self.address_offset
+                )
+                row.parameter_uuid = table_element2.uuid
+                parameter = self.parameter_uuid_finder(table_element2.uuid)
+                uses_interface_item = (
+                    isinstance(parameter, epyqlib.pm.parametermodel.TableArrayElement)
+                    and parameter.uses_interface_item()
+                )
+                row.parameter_uses_interface_item = uses_interface_item
+
+                # TODO: the scale factor is on the TableRepeatingBlockReference children, so something special will need to be done.
+                # if row.scale_factor is not None:
+                #     row.scale_factor_uuid = row.scale_factor
+                #     row.scale_factor = self.parameter_uuid_finder(
+                #         self.scale_factor_from_uuid[row.scale_factor].parameter_uuid
+                #     ).abbreviation
+
+                row.enumeration_uuid = table_element2.enumeration_uuid
+                row.not_implemented = point.not_implemented
+                row.uuid = point.uuid
+                row.class_name = "TableRepeatingBlockReferenceDataPointReference"
+
+                rows.append(row)
+
+                # Increase the address offset by the size of the data type.
+                block_size += point.size
+                curve_points += 1
+
+        return rows, block_size
+
+
+@builders(epcpm.sunspecmodel.TableRepeatingBlock)
+@attr.s
+class TableRepeatingBlock:
+    wrapped = attr.ib()
+    add_padding = attr.ib()
+    padding_type = attr.ib()
+    model_type = attr.ib()
+    model_id = attr.ib()
+    model_offset = attr.ib()
+    address_offset = attr.ib()
+    repeating_block_reference = attr.ib(default=None)
+    parameter_uuid_finder = attr.ib(default=None)
+    is_table = attr.ib(default=False)
+
+    def gen(self):
+        return [], 0
+
+
+@builders(epcpm.sunspecmodel.DataPoint)
+@attr.s
+class Point:
+    wrapped = attr.ib()
+    scale_factor_from_uuid = attr.ib()
+    model_type = attr.ib()
+    model_id = attr.ib()
+    model_offset = attr.ib()
+    is_table = attr.ib()
+    address_offset = attr.ib()
+    repeating_block_reference = attr.ib()
+    parameter_uuid_finder = attr.ib(default=None)
+
+    # TODO: CAMPid 07397546759269756456100183066795496952476951653
```

Comment:
Why this TODO?

Reply:
I believe the idea behind TODO comments like this is that when a code snippet has been copied from elsewhere, that this TODO identifies it in order for a future developer to refactor. The string uniquely identifies the code snippet, so there should be at least one other code snippet like the one above. There are many examples of TODO comments like this one, to the point where it would take a very large effort to refactor.

Reply:
Interesting, thanks for the explanation.